code/framework/object/src/main/java/io/cattle/platform/object/meta/impl/DefaultObjectMetaDataManager.java

```diff
@@ -683,6 +683,25 @@ public Relationship getRelationship(String type, String linkName) {
     }
 
     @Override
+    public Relationship getRelationship(String type, String linkName, String fieldName) {
+        Class<?> clz = schemaFactory.getSchemaClass(type, true);
+        Map<String, List<Relationship>> relationship = relationshipsBothCase.get(clz);
+        List<Relationship> relationshipList = relationship.get(linkName);
+        if (relationshipList == null || relationshipList.size() == 0) {
+            return null;
+        }
+        if (relationshipList.size() > 1) {
+            for (Relationship rel : relationshipList) {
+                if (rel instanceof ForeignKeyRelationship && rel.getName() != null && rel.getName().equalsIgnoreCase(linkName) && rel.getPropertyName().equalsIgnoreCase(fieldName)) {
+                    return rel;
+                }
+            }
+            return null;
+        }
```

Comment:
Its not the implementation; @sonchang modified getRelationShip() method itself in DefaultObjectMetaDataManager.java with commit 4f9332222b866fce9be057dc04b2140edd59af37. His fix was to implement handling more than 1 relationship having the same link name.